test/integration/integration_test.py

```diff
@@ -0,0 +1,85 @@
+from contextlib import contextmanager
+from marquez_client.marquez import MarquezClient
+from marquez_codegen_client.rest import ApiException
+from time import sleep
+from urllib3.exceptions import (MaxRetryError,
+                                NewConnectionError,
+                                ReadTimeoutError)
+
+import pytest
+import requests
+import subprocess
+import sys
+
+MARQUEZ_HOST = "marquez"
+MARQUEZ_PORT = "5000"
+
+
+def test_bad_host():
+    assert wait_for_marquez()
+    c = MarquezClient(host="bad-host", port=MARQUEZ_PORT)
+    with pytest.raises(MaxRetryError) as e:
+        c.get_namespace_info("no_connection")
+    assert isinstance(e.value.reason, NewConnectionError)
+
+
+def test_bad_port():
+    assert wait_for_marquez()
+    c = MarquezClient(host=MARQUEZ_HOST, port="6000")
+    with pytest.raises(MaxRetryError) as e:
+        c.get_namespace_info("no_connection")
+    assert isinstance(e.value.reason, NewConnectionError)
+
+
+def test_timeout():
+    assert wait_for_marquez()
+    c = MarquezClient(host=MARQUEZ_HOST, port=MARQUEZ_PORT, timeout=1)
+
+    with broken_network():
+        expected_namespace = "timeout_test"
+        with pytest.raises(MaxRetryError) as e:
+            c.get_namespace_info(expected_namespace)
+        assert isinstance(e.value.reason, ReadTimeoutError)
+
+
+def test_namespace_not_found():
+    assert wait_for_marquez()
+    c = MarquezClient(host=MARQUEZ_HOST, port=MARQUEZ_PORT)
+
+    expected_namespace = "not_found"
+    with pytest.raises(ApiException):
+        c.get_namespace_info(expected_namespace)
+
+
+@contextmanager
+def broken_network():
+    iptables_drop_packets(True)
+    yield
+    iptables_drop_packets(False)
+
+
+def iptables_drop_packets(drop):
+    subprocess.run(["iptables",
+                    "-A" if drop else "-D", "INPUT",
+                    "-p", "tcp",
+                    "--tcp-flags", "PSH", "PSH",
+                    "--sport", MARQUEZ_PORT,
+                    "-j", "DROP"])
+
+
+def wait_for_marquez(retries=10):
+    while(retries):
+        try:
+            r = requests.get('http://{}:{}/ping'.format(MARQUEZ_HOST,
+                                                        MARQUEZ_PORT))
+            if r.status_code == 200:
```

Comment:
nit: Please use the httplib constants: https://docs.python.org/2/library/httplib.html

Reply:
will do